bacom-upload/migration.js

```diff
@@ -0,0 +1,277 @@
+/* eslint-disable max-len */
+import fs from 'fs';
+import readline from 'readline';
+import dotenv from 'dotenv';
+import { u } from 'unist-builder';
+import { fetch } from '@adobe/fetch';
+import { mdast2docx } from '@adobe/helix-md2docx';
+import { BulkUpdate, ExcelReporter, loadListData, saveDocument } from '../bulk-update/index.js';
+import { selectBlock } from '../bulk-update/migration-tools/select.js';
+import { localizeStagePath } from '../bulk-update/bulk-update.js';
+import { entryToPath } from '../bulk-update/document-manager/document-manager.js';
+
+dotenv.config({ path: 'bacom-upload/.env' });
+
+const { SITE_ID, DRIVE_ID, BEARER_TOKEN } = process.env;
+const GRAPH_UPLOAD = true;
+const { pathname } = new URL('.', import.meta.url);
+const dateString = ExcelReporter.getDateString();
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const config = {
+  list: [
+    'https://main--bacom--adobecom.hlx.live/query-index.json',
+  ],
+  siteUrl: 'https://main--bacom--adobecom.hlx.live',
+  stagePath: '/drafts/staged-content',
+  locales: JSON.parse(fs.readFileSync(`${pathname}locales.json`, 'utf8')),
+  prodSiteUrl: 'https://business.adobe.com',
+  reporter: new ExcelReporter(`${pathname}reports/${dateString}.xlsx`, false),
+  outputDir: `${pathname}output`,
+  mdDir: `${pathname}md`,
+  mdCacheMs: 1 * 24 * 60 * 60 * 1000, // 1 day(s)
+  fetchWaitMs: 20,
+};
+
+/**
+ * Creates a block with the given name and fields.
+ *
+ * @param {string} name - The name of the block.
+ * @param {Object} fields - The fields of the block.
+ * @returns {Array} - The created block.
+ */
+export function createBlock(name, fields) {
+  const block = u('gridTable', [
+    u('gtBody', [
+      u('gtRow', [
+        u('gtCell', { colSpan: 2 }, [u('paragraph', [u('text', name)])]),
+      ]),
+      ...fields.map((values) => u('gtRow', values.map((value) => u('gtCell', [u('paragraph', [value ?? u('text', '')])])))),
+    ]),
+  ]);
+
+  return block;
+}
+
+/**
+ * Uploads a chunk of data to the specified upload URL.
+ *
+ * @param {string} uploadUrl - The URL to upload the chunk to.
+ * @param {Buffer} buffer - The buffer containing the data to be uploaded.
+ * @param {number} chunkStart - The start index of the chunk in the buffer.
+ * @param {number} chunkEnd - The end index of the chunk in the buffer.
+ * @returns {Promise<Object|boolean>} - A promise that resolves to the upload data if successful, or false if an error occurs.
+ */
+async function uploadChunk(uploadUrl, buffer, chunkStart, chunkEnd) {
+  const headers = { 'Content-Range': `bytes ${chunkStart}-${chunkEnd - 1}/${buffer.length}` };
+  const body = buffer.slice(chunkStart, chunkEnd);
+
+  try {
+    const res = await fetch(uploadUrl, { method: 'PUT', body, headers });
+    console.log(res);
+    const uploadData = await res.json();
+    console.log(uploadData);
```

Comment:
Are you keeping these console.logs in?

Reply:
No, I'm going to replace this with a report.log with the webUrl to the Sharepoint edit link.